pymc_extras/statespace/models/structural/core.py

```diff
@@ -764,15 +806,17 @@ def _combine_property(self, other, name, allow_duplicates=True):
         self_prop = getattr(self, name)
         other_prop = getattr(other, name)
 
+        # TODO discuss limiting the types we get here to only a dataclass type. By making the dataclasses immutable we now have to handle for tuples too.
+
         if not isinstance(self_prop, type(other_prop)):
             raise TypeError(
                 f"Property {name} of {self} and {other} are not the same and cannot be combined. Found "
                 f"{type(self_prop)} for {self} and {type(other_prop)} for {other}'"
             )
 
-        if not isinstance(self_prop, list | dict):
+        if not is_dataclass(self_prop) and not isinstance(self_prop, list | tuple | dict):
             raise TypeError(
-                f"All component properties are expected to be lists or dicts, but found {type(self_prop)}"
+                f"All component properties are expected to be dataclasses, but found {type(self_prop)}"
```

Comment:
Would be better to have this error say the specific class it expects (ParameterInfo / DataInfo / etc) rather than just generic "dataclass"